demo/src/features/mesh/jobs/prepare.rs

```diff
@@ -297,58 +284,172 @@ impl<'prepare> PrepareJobEntryPoints<'prepare> for MeshPrepareJob<'prepare> {
             }
 
             let mesh_part = mesh_part.as_ref().unwrap();
-            let textured_descriptor_set = if self.requires_textured_descriptor_sets {
-                let per_object_data = MeshPerObjectFragmentShaderParam { model };
-
-                let per_instance_descriptor_set_layout =
-                    &mesh_part.material_instance.material.passes[mesh_part.textured_pass_index]
-                        .material_pass_resource
-                        .get_raw()
-                        .descriptor_set_layouts
-                        [PER_INSTANCE_DESCRIPTOR_SET_INDEX as usize];
-
-                descriptor_set_allocator
-                    .create_descriptor_set_with_writer(
-                        per_instance_descriptor_set_layout,
-                        shaders::mesh_textured_frag::DescriptorSet2Args {
-                            per_object_data: &per_object_data,
-                        },
-                    )
-                    .ok()
-            } else {
-                None
-            };
 
-            let untextured_descriptor_set = if self.requires_untextured_descriptor_sets {
-                let per_object_data = MeshPerObjectFragmentShaderParam { model };
-
-                let per_instance_descriptor_set_layout =
-                    &mesh_part.material_instance.material.passes[mesh_part.untextured_pass_index]
-                        .material_pass_resource
-                        .get_raw()
-                        .descriptor_set_layouts
-                        [PER_INSTANCE_DESCRIPTOR_SET_INDEX as usize];
-
-                descriptor_set_allocator
-                    .create_descriptor_set_with_writer(
-                        per_instance_descriptor_set_layout,
-                        shaders::mesh_textured_frag::DescriptorSet2Args {
-                            per_object_data: &per_object_data,
-                        },
-                    )
-                    .ok()
-            } else {
-                None
-            };
+            if let Some(dyn_material_descriptor_set_index) = dyn_material_descriptor_set_index {
+                let extracted_mesh_part_index =
+                    extracted_data.dyn_mesh_material_offset.unwrap() + mesh_part_index;
+                let descriptor_sets = if let Some(extracted_dyn_mesh_material) =
+                    self.dyn_mesh_materials.get(extracted_mesh_part_index)
+                {
+                    let material = &extracted_dyn_mesh_material.per_material_data;
+
+                    job_context
+                        .dyn_material_descriptor_sets
+                        .entry(material.clone())
+                        .or_insert_with(|| {
+                            // Dyn material descriptor sets.
+
+                            let textured_descriptor_set = {
+                                let per_material_data =
+                                    shaders::mesh_textured_frag::MaterialDataUboUniform {
```

Comment:
The duplication of this uniform just because the `mesh_textured_frag` becomes `mesh_untextured_frag` is annoying. Maybe just `transmute` it since it's the _same_ struct?

Reply:
Punting on dynamic materials.